specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,236 @@
+
+    @doc("Identifies the context in which an event happened. The combination of id and source must be unique for each distinct event.")
+    source: string;
+
+    @doc("Event data specific to the event type.")
+    data?: object;
+
+    @doc("Event data specific to the event type, encoded as a base64 string.")
+    data_base64?: bytes;
+    
+    @doc("Type of event related to the originating occurrence.")
+    type: string;
+
+    @doc("The time (in UTC) the event was generated, in RFC3339 format.")
+    time?: zonedDateTime;
+
+    @doc("The version of the CloudEvents specification which the event uses.")
+    specversion: string;
+
+    @doc("Identifies the schema that data adheres to.")
+    dataschema?: string;
+
+    @doc("Content type of data value.")
+    datacontenttype?: string;
+
+    @doc("This describes the subject of the event in the context of the event producer (identified by source).")
+    subject?: string;
+  }
+
+  @doc("LockToken information.")
+  model LockToken {
+    @doc("The token used to lock the event.")
+    lockToken: string;
+  }
+
+  @doc("Properties of the Event Broker operation.")
+  model BrokerProperties {
+    @doc("The token used to lock the event.")
+    lockToken: LockToken;
+  }
+
+  @doc("Receive operation details per Cloud Event.") 
+  model ReceiveDetails {
+    @doc("The Event Broker details.")
+    brokerProperties: BrokerProperties;
+
+    @doc("Cloud Event details.")
+    event: CloudEventEvent;
+  }
+
+  @doc("Details of the Receive operation response.")
+  model ReceiveResponse {
+    @doc("Array of receive responses, one per cloud event.")
+    value: ReceiveDetails[];
+  }
+   
+  @doc("Failed LockToken information.")
+  model FailedLockToken {
+    @doc("LockToken value")
+    lockToken: LockToken;
+
+    @doc("Error code")
+    errorCode: int32;
+
+    @doc("Description of the error")
+    errorDescription: string; 
+  }
+   
+  @doc("Details of the LockTokens information. This is used for both Acknowledge and Release operation response.")
+  model LockTokensResponse {
+    @doc("Array of LockToken values for failed cloud events.")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of LockToken values for succeeded cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  @doc("Lock token input formatting.")
+  model LockTokenInput {
+    @doc("LockToken")
+    lockTokens: string[];
+  }
+
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+
+  @doc("Publish Single Cloud Event to namespace topic.")
+  @route("/topics/{topicName}:publish", {shared: true})
+  @post op PublishCloudEvent is Azure.Core.RpcOperation<{
+    @doc("content type")
+    @header("content-type")
+    contentType: "application/cloudevents+json; charset=utf-8";
+
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Single Cloud Event being published.")
+    @body 
+    event: CloudEventEvent;
+  }, OkResponse>;
```

Comment:
From testing the server, we indeed receive a 200. We would have expected a 204, not a 200, given this returns no body.